webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -103,6 +135,703 @@ object RenderInheritedProperties {
+      ram:                         Option[MemorySize],
+      machine:                     Option[JRNodeDetailLevel.MachineInfo],
+      ipAddresses:                 Option[Chunk[String]],
+      description:                 Option[String],
+      lastInventoryDate:           Option[DateTime],
+      lastRunDate:                 Option[DateTime],
+      policyServerId:              Option[NodeId],
+      managementTechnology:        Option[Chunk[JRNodeDetailLevel.Management]],
+      properties:                  Option[Chunk[JRProperty]],
+      policyMode:                  Option[String],
+      timezone:                    Option[JRNodeDetailLevel.NodeTimezone],
+      tenant:                      Option[TenantId],
+      // full
+      accounts:                    Option[Chunk[String]],
+      bios:                        Option[Chunk[JRNodeDetailLevel.Bios]],
+      controllers:                 Option[Chunk[JRNodeDetailLevel.Controller]],
+      environmentVariables:        Option[Map[String, String]],
+      fileSystems:                 Option[Chunk[JRNodeDetailLevel.FileSystem]],
+      managementTechnologyDetails: Option[JRNodeDetailLevel.ManagementDetails],
+      memories:                    Option[Chunk[JRNodeDetailLevel.MemorySlot]],
+      networkInterfaces:           Option[Chunk[JRNodeDetailLevel.Network]],
+      processes:                   Option[Chunk[JRNodeDetailLevel.Process]],
+      processors:                  Option[Chunk[JRNodeDetailLevel.Processor]],
+      slots:                       Option[Chunk[JRNodeDetailLevel.Slot]],
+      software:                    Option[Chunk[JRNodeDetailLevel.Software]],
+      softwareUpdate:              Option[Chunk[SoftwareUpdate]],
+      sound:                       Option[Chunk[JRNodeDetailLevel.Sound]],
+      storage:                     Option[Chunk[JRNodeDetailLevel.Storage]],
+      ports:                       Option[Chunk[JRNodeDetailLevel.Port]],
+      videos:                      Option[Chunk[JRNodeDetailLevel.Video]],
+      virtualMachines:             Option[Chunk[JRNodeDetailLevel.VirtualMachine]]
+  )
+
+  object JRNodeDetailLevel {
+    implicit def nodeFactTransformer(implicit
+        nodeFact: NodeFact,
+        status:   InventoryStatus,
+        agentRun: Option[AgentRunWithNodeConfig]
+    ): Transformer[NodeDetailLevel, JRNodeDetailLevel] = {
+      implicit val nodeInfo:    NodeInfo               = nodeFact.toNodeInfo
+      implicit val securityTag: Option[SecurityTag]    = nodeFact.rudderSettings.security
+      implicit val software:    Chunk[domain.Software] = nodeFact.software.map(_.toSoftware)
+      // we could keep inventory as Option to make syntax easier to filter empty lists, or write Some(...).filter(_.nonEmpty) everywhere
+      implicit val inventory:   Option[FullInventory]  = Some(nodeFact.toFullInventory)
+      transformer
+    }
+
+    implicit def transformer(implicit
+        nodeInfo:    NodeInfo,
```

Comment:
you need to start from `NodeFact` here. We kept nodeinfo / inventory for compat reason with the existing API, but that's not the goal. We need to take the opportunity to remove it enterily from here.

Reply:
:white_check_mark: => https://github.com/Normation/rudder/pull/5656/commits/b8c2bf3e57e6486c5fb19c633a0c75199e84271f